Rules.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
+
+namespace TicTacToe
+{
+    public class Rules
+    {
+        public bool IsWon(List<object> board, char playerSymbol)
+        {
+            if (board[0].Equals(playerSymbol) && board[1].Equals(playerSymbol) && board[2].Equals(playerSymbol)) { return true; }
+            if (board[3].Equals(playerSymbol) && board[4].Equals(playerSymbol) && board[5].Equals(playerSymbol)) { return true; }
+            if (board[6].Equals(playerSymbol) && board[7].Equals(playerSymbol) && board[8].Equals(playerSymbol)) { return true; }
+
+            if (board[0].Equals(playerSymbol) && board[3].Equals(playerSymbol) && board[6].Equals(playerSymbol)) { return true; }
+            if (board[1].Equals(playerSymbol) && board[4].Equals(playerSymbol) && board[7].Equals(playerSymbol)) { return true; }
+            if (board[2].Equals(playerSymbol) && board[5].Equals(playerSymbol) && board[8].Equals(playerSymbol)) { return true; }
+
+            if (board[0].Equals(playerSymbol) && board[4].Equals(playerSymbol) && board[8].Equals(playerSymbol)) { return true; }
+            if (board[2].Equals(playerSymbol) && board[4].Equals(playerSymbol) && board[6].Equals(playerSymbol)) { return true; }

```

Comment:
There is a lot of duplication here! One way to approach refactoring is to think about how you can generalize the winning logic for square boards of any dimension.

Reply:
Hmm~ I came up with some sort of logic, but I'm not sure if this was what you were looking for. After writing down all the winning combinations of a 2x2, 3x3, 4x4 board, assuming `n = grid dimension`, having 2 nested for loops, `i` &` j` where it starts at 0 and increments by 1 until less than `n`:

Horizontal wins can be calculated `(n * j) + i`

Vertical wins can be calculated `(n * i) + j`

..but I'm not sure if I picked up the diagonals correctly. 
For a left diagonal, I noticed `(n + 1) * i` works
For a right diagonal, I noticed `(n - 1) * (i + 1)` works

For the purposes of this TTT, since I am only working on a 3 x 3 board, I hardcoded the diagonals, but I'm still curious as to what pattern you were referring to. Please enlighten me 🙏 !

Here's my first attempt: e8f0227

Reply:
That's more than fine! I didn't have something specific in mind, just wanted to see less duplication and hard-coded values. I really like the refactor, and I think it's fine to keep the diagonals hard-coded for now, given the requirements.

A couple of nitpicks: 
```java
if (tempArr[0].Equals(playerSymbol) && tempArr[1].Equals(playerSymbol) && tempArr[2].Equals(playerSymbol)) return true;
```
1. You should be able to just `return` the boolean expression directly instead of using `if` here.
2. This expression is duplicated in the horizontal, vertical and the two diagonals, so it might be worth extracting into a separate function.

Reply:
😃 addressed in 0a419c6